builtin/providers/aws/resource_aws_instance.go

```diff
@@ -491,10 +496,12 @@ func resourceAwsInstanceRead(d *schema.ResourceData, meta interface{}) error {
 		for _, ni := range instance.NetworkInterfaces {
 			if *ni.Attachment.DeviceIndex == 0 {
 				d.Set("subnet_id", ni.SubnetId)
+				d.Set("network_interface_id", ni.NetworkInterfaceId)
 			}
 		}
 	} else {
 		d.Set("subnet_id", instance.SubnetId)
+		d.Set("network_interface_id", "")
```

Comment:
Because there are no network interfaces to get it from. If there is one from somewhere else that I don't know about, I'd be happy to change it.